### DISCUSSION TITLE
ResourceHandler sometimes fails when open the browser for the first time?

### DISCUSSION THREAD

**Discussion by tp1415926535** [cefsharp/CefSharp#3694] [General]
I inherited **ResourceHandler** and **ISchemeHandlerFactory** classes, set **RegisterScheme** in app class to get **embedded** resources(html and js files), and use **ExecuteScriptAsyncWhenPageLoaded** to call js functions, 
But when I show the browser window(child window) for the first time, it sometimes throw error that some functions or variables are undefined. Then I close this child window and show again, it will never report an error again and successfully display the page ,execute the function . This is the case in most cases when I test it.
So I want to know where I might be wrong?

**Comment by amaitland** [comment]
Please provide
- the exact error message you are seeing
- code example that reproduces the problem 

Please fork the minimum example, make only the minimal changes required to reproduce the problem, squash all changes into a single commit, push changes to GitHub and post a link here to your fork.

> So I want to know where I might be wrong?

Without an example it's impossible to say. See https://stackoverflow.com/help/minimal-reproducible-example

**Reply by tp1415926535** [reply]
I found similar question [here](https://stackoverflow.com/questions/61749101/cefsharp-uncaught-referenceerror-method-is-not-defined-n-aboutblank10) and referenced the [documentation](https://github.com/cefsharp/CefSharp/wiki/General-Usage#when-can-i-start-executing-javascript) mentioned in the recommendations. And then I put my **Browser.ExecuteScriptAsyncWhenPageLoaded** event in to **Browser.LoadingStateChanged** method , the probability of error seems to have been reduced. So, I don't know why ExecuteScriptAsyncWhenPageLoaded sometimes happened before resources loaded, although it seems that this still happens sometimes.
